### PR TITLE
Upgrade to analyzer 0.34.0

### PR DESCRIPTION
**Pull request by stereotype441**

### CODE REVIEW

**Comment by pq** [comment]
Looks like `dart_style` needs an update before this can land?

https://travis-ci.org/dart-lang/linter/jobs/461514903

```
Because dart_style >=0.1.1 <1.1.3 requires SDK version <2.0.0 or >=2.0.0-dev.17.0 <2.0.0 and dart_style >=1.1.3 <1.2.0 depends on analyzer >0.31.2-alpha.0 <0.33.0, dart_style >=0.1.1 <1.2.0 requires analyzer >0.31.2-alpha.0 <0.33.0.
And because dart_style >=1.2.0 depends on analyzer ^0.33.0-alpha.0, dart_style >=0.1.1 requires analyzer >0.31.2-alpha.0 <0.33.0 or ^0.33.0-alpha.0.
So, because linter depends on both analyzer ^0.34.0 and dart_style ^1.1.0, version solving failed.
```

**Comment by pq** [comment]
@stereotype441: maybe updating the `dart_style` constraint to `^1.2.0` will do it?

**Comment by stereotype441** [comment]
Hmm, looks like dart_style needs to be rolled to 0.34.0 first.  I'll go make a PR to do that.

File: test/mock_sdk.dart [hunk 1]
```diff
@@ -142,6 +142,10 @@ abstract class Map<K, V> extends Object {
   V putIfAbsent(K key, V ifAbsent());
 }
 
+abstract class Set<E> implements Iterable<E> {
+  Set<R> cast<R>();
+}
+
```
Comment (pq) [inline test/mock_sdk.dart]: Awesome.  Thanks!

**Pull request merged**